internal/git/git.go

```diff
@@ -139,6 +140,10 @@ func (g *GitOperations) GetBranchShas(ctx context.Context, branch string) (Branc
 	metadataFile := gitpaths.Get(g.gap.GetGitHttpsRepoUrl(*g.gitRepo)+g.pathContext) + "/hydrator.metadata"
 	jsonFile, err := os.Open(metadataFile)
 	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
```

Comment:
Needed since https://github.com/argoproj-labs/gitops-promoter/pull/250 was merged because the error is not handled anymore.